File: src/main/scala/net/machinemuse/utils/MuseFileUtils.scala
package net.machinemuse.utils

import java.io.{File, FileInputStream, FileOutputStream}

import net.machinemuse.numina.general.MuseLogger

/**
 * Author: MachineMuse (Claire Semple)
 * Created: 11:07 PM, 6/28/13
 */
object MuseFileUtils {
  def copyFile(oldFile: File, newFile: File) {
    if (!newFile.exists) {
      newFile.createNewFile()
    }
    new FileOutputStream(newFile).getChannel transferFrom(new FileInputStream(oldFile).getChannel, 0, Long.MaxValue)
    MuseLogger.logDebug("Successfully moved MPS config to new location. :D")
  }
}

File: src/main/java/net/machinemuse/utils/MuseFileUtils.java
package net.machinemuse.utils;

import net.machinemuse.numina.utils.MuseLogger;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;

/**
 * Author: MachineMuse (Claire Semple)
 * Created: 11:07 PM, 6/28/13
 *
 * Ported to Java by lehjr on 10/11/16.
 */
public class MuseFileUtils {
    public void copyFile(File oldFile, File newFile) throws IOException {
        if (!newFile.exists()) {
            newFile.createNewFile();
        }
        new FileOutputStream(newFile).getChannel().transferFrom(new FileInputStream(oldFile).getChannel(), 0, Long.MAX_VALUE);
        MuseLogger.logDebug("Successfully moved MPS config to new location. :D");
    }
}


